src/components/feature-board/content-elements/FileContentElementMenu.unit.ts

```diff
@@ -0,0 +1,399 @@
+				.findAllComponents(BoardMenuAction)
+				.filter((c) => c.text().includes(deleteTranslation))
+				.at(0);
+
+			expect(childComponent.exists()).toBe(true);
+		});
+
+		it("should find download board menu action", () => {
+			const { wrapper } = setup();
+
+			const downloadTranslation = wrapper.vm
+				.$t("components.board.action.download")
+				.toString();
+			const childComponent = wrapper
+				.findAllComponents(BoardMenuAction)
+				.filter((c) => c.text().includes(downloadTranslation))
+				.at(0);
+
+			expect(childComponent.exists()).toBe(true);
+		});
+
+		describe("when delete board menu action is clicked", () => {
+			it("should emit delete:element event", async () => {
+				const { wrapper } = setup();
+
+				const deleteTranslation = wrapper.vm
+					.$t("components.board.action.delete")
+					.toString();
+				const childComponent = wrapper
+					.findAllComponents(BoardMenuAction)
+					.filter((c) => c.text().includes(deleteTranslation))
+					.at(0);
+				childComponent.vm.$emit("click");
+
+				expect(wrapper.emitted("delete:element")?.length).toBe(1);
+			});
+		});
+
+		describe("when download board menu action is clicked", () => {
+			const setup = () => {
+				document.body.setAttribute("data-app", "true");
+
+				const downloadFileMock = jest
+					.mocked(downloadFile)
+					.mockReturnValueOnce();
+
+				const propsData = setupProps();
+				const wrapper = shallowMount(FileContentElementMenu, {
+					...createComponentMocks({ i18n: true }),
+					propsData,
+					provide: {
+						[I18N_KEY as symbol]: { t: (key: string) => key },
+					},
+				});
+
+				return {
+					wrapper,
+					fileNameProp: propsData.fileName,
+					urlProp: propsData.url,
+					downloadFileMock,
+				};
+			};
+
+			it("should download file", async () => {
+				const { wrapper, urlProp, fileNameProp, downloadFileMock } = setup();
+
+				const downloadTranslation = wrapper.vm
+					.$t("components.board.action.download")
+					.toString();
+				const childComponent = wrapper
+					.findAllComponents(BoardMenuAction)
+					.filter((c) => c.text().includes(downloadTranslation))
+					.at(0);
+
+				childComponent.vm.$emit("click");
+
+				expect(downloadFileMock).toHaveBeenCalledTimes(1);
+				expect(downloadFileMock).toHaveBeenCalledWith(urlProp, fileNameProp);
+			});
+		});
+	});
+
+	describe("when multiple elements are present", () => {
+		const setup = () => {
+			document.body.setAttribute("data-app", "true");
+
+			const multipleElementsSetupProps = {
+				fileId: "file-id #1",
+				fileName: "file-record #1.txt",
+				url: "1/file-record #1.txt",
+				isDownloadAllowed: true,
+				isFirstElement: false,
+				isLastElement: false,
+				hasMultipleElements: true,
+			};
+			const wrapper = shallowMount(FileContentElementMenu, {
+				...createComponentMocks({ i18n: true }),
+				propsData: multipleElementsSetupProps,
+			});
+
+			return {
+				wrapper,
+			};
+		};
+
+		it("should show all board menu items", () => {
```

Comment:
The download and delete action are independent of the number of elements. So I wouldn't mention them here and call this 
```suggestion
		it("should show the move up and move down action", () => {
```
Same for the tests below.

Reply:
Fixed by restructuring `FileContentElement` unit tests! Separated different menu actions (`move up/down`, `download`, and `delete`) into separate scenarios.